### PR TITLE
Rename Options Recognizers to Choice Recognizers and Port it to C#

### PR DESCRIPTION
**Pull request by amitstein**

### CODE REVIEW

File: Specs/Options/English/BooleanModel.json [hunk 1]
```diff
@@ -60,7 +60,7 @@
 ,
 {
   "Input": "yes... I said no onions",
-  "NotSupportedByDesign": "python",
+  "NotSupportedByDesign": "python, dotNet",
```
Comment (tellarin) [inline Specs/Options/English/BooleanModel.json]: Why is this the case? Shouldn't it behave like the javascript version?

File: .NET/Samples/SimpleConsole_Core/Program.cs [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Recognizers.Text;
+using Microsoft.Recognizers.Text.DateTime;
+using Microsoft.Recognizers.Text.Number;
+using Microsoft.Recognizers.Text.NumberWithUnit;
+using Microsoft.Recognizers.Text.Sequence;
+using Newtonsoft.Json;
+
+namespace SimpleConsole_Core
```
Comment (tellarin) [inline .NET/Samples/SimpleConsole_Core/Program.cs]: Do we really need two SimpleConsole? I think we can keep just one, and that one can be a .NET core project, as we want to make sure the .NET standard version of the recognizers is the preferred target.

Comment (tellarin) [reply .NET/Samples/SimpleConsole_Core/Program.cs]: Please make another small PR with just this change, as it's unrelated to the new recognizer.

File: .NET/Microsoft.Recognizers.Definitions/English/OptionsDefinitions.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+namespace Microsoft.Recognizers.Definitions.English
+{
+	using System;
+	using System.Collections.Generic;
+
+	public static class OptionsDefinitions
+	{
+		public const string LangMarker = "Eng";
+		public const string TokenizerRegex = @"[^\w\d]";
+		public const string TrueRegex = @"(true|yes|yep|y|sure|ok|\uD83D\uDC4D|\uD83D\uDC4C)";
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Definitions/English/OptionsDefinitions.cs]: Shouldn't we enforce word boundaries here?

File: .NET/Microsoft.Recognizers.Text.Options/Extractors/IChoiceExtractorConfiguration.cs [hunk 4]
```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+namespace Microsoft.Recognizers.Text.Options.Extractors
+{
+    public interface IChoiceExtractorConfiguration
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.Options/Extractors/IChoiceExtractorConfiguration.cs]: I see that in some places structure classes are named "Choice" and in others "Options". Let's rename all for consistency once the discussion on the best name converges to one.

File: .NET/Microsoft.Recognizers.Text.Options/Extractors/OptionsExtractor.cs [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+                var initialScore = accuracy * (matched / source.Count());
+
+                score = 0.4 + (0.6 * initialScore);
+            }
+            return score;
+        }
+
+        private static int IndexOfToken(List<string> tokens, string token, int startPos)
+        {
+            if (tokens.Count <= startPos) return -1;
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.Options/Extractors/OptionsExtractor.cs]: Nit: Code standard. Please add the {}

File: Specs/Options/English/BooleanModel.json (hunk 1 above)
Comment (amitstein) [reply Specs/Options/English/BooleanModel.json]: Removed "dotnet not supported"

File: .NET/Samples/SimpleConsole_Core/Program.cs (hunk 2 above)
Comment (amitstein) [reply .NET/Samples/SimpleConsole_Core/Program.cs]: Sample removed

File: .NET/Microsoft.Recognizers.Definitions/English/OptionsDefinitions.cs (hunk 3 above)
Comment (amitstein) [reply .NET/Microsoft.Recognizers.Definitions/English/OptionsDefinitions.cs]: Added boundaries only for words (not for emoji)

File: .NET/Microsoft.Recognizers.Text.Options/Extractors/IChoiceExtractorConfiguration.cs (hunk 4 above)
Comment (amitstein) [reply .NET/Microsoft.Recognizers.Text.Options/Extractors/IChoiceExtractorConfiguration.cs]: Renamed Options recognizer to Choice recognizer

File: Specs/Options/English/BooleanModel.json (hunk 1 above)
Comment (amitstein) [reply Specs/Options/English/BooleanModel.json]: Skip test removed

File: .NET/Microsoft.Recognizers.Text.Options/Extractors/OptionsExtractor.cs (hunk 5 above)
Comment (amitstein) [reply .NET/Microsoft.Recognizers.Text.Options/Extractors/OptionsExtractor.cs]: Fixed

**Comment by tellarin** [comment]
Hi @amitstein,
there are a few conflicts now that we merged the API signature changes PR. Can you please resolve them? 
Thanks!

File: .NET/Microsoft.Recognizers.Text.Options/Extractors/IChoiceExtractorConfiguration.cs (hunk 4 above)
Comment (tellarin) [reply .NET/Microsoft.Recognizers.Text.Options/Extractors/IChoiceExtractorConfiguration.cs]: 👍

**Pull request merged**